core/domain/blog_services.py

```diff
@@ -0,0 +1,613 @@
+            blog_post_summary_model)
+        return blog_post_summary
+    else:
+        return None
+
+
+def get_blog_post_summary_models_list_by_user_id(
+        user_id, blog_post_is_published):
+    """Given the user ID and status, it returns the list of blog post summary
+    domain object for which user is an editor and the status matches.
+
+    Args:
+        user_id: str. The user who is editor of the blog posts.
+        blog_post_is_published: bool. Whether the given blog post is
+            published or not.
+
+    Returns:
+        list(BlogPostSummary). The blog post summaries of the blog posts for
+        which the user is an editor corresponding to the status
+        (draft/published).
+    """
+    blog_post_ids = filter_blog_post_ids(user_id, blog_post_is_published)
+    blog_post_summary_models = (
+        blog_models.BlogPostSummaryModel.get_multi(blog_post_ids))
+    blog_post_summaries = []
+    blog_post_summaries = [
+        get_blog_post_summary_from_model(model) if model is not None else None
+        for model in blog_post_summary_models]
+    return blog_post_summaries if len(blog_post_summaries) != 0 else None
+
+
+def filter_blog_post_ids(user_id, blog_post_is_published):
+    """Given the user ID and status, it returns the IDs of all blog post
+    according to the status.
+
+    Args:
+        user_id: str. The user who is editor of the blog post.
+        blog_post_is_published: bool. True if blog post is published.
+
+    Returns:
+        list(str). The blog post IDs of the blog posts for which the user is an
+        editor corresponding to the status(draft/published).
+    """
+    blog_post_rights_models = (
+        blog_models.BlogPostRightsModel.get_by_user(user_id))
+    model_ids = []
+    if blog_post_rights_models:
+        for model in blog_post_rights_models:
+            if model.blog_post_is_published == blog_post_is_published:
+                model_ids.append(model.id)
+    return model_ids
+
+
+def get_blog_post_summary_by_title(title):
+    """Returns a domain object representing a blog post summary model.
+
+    Args:
+        title: str. The title of the blog post.
+
+    Returns:
+        BlogPostSummary or None. The domain object representing a blog post
+        summary with the given title, or None if it does not exist.
+    """
+    blog_post_summary_model = blog_models.BlogPostSummaryModel.query(
+        blog_models.BlogPostSummaryModel.title == title
+    ).fetch() # pylint: disable=singleton-comparison
+
+    if len(blog_post_summary_model) == 0:
+        return None
+
+    return get_blog_post_summary_from_model(blog_post_summary_model[0])
+
+
+def get_all_blog_posts():
+    """Returns all the blog posts present in the datastore.
+
+    Returns:
+        list(BlogPosts). The list of domain objects for blog posts present
+        in the datastore.
+    """
+    backend_blog_post_models = blog_models.BlogPostModel.get_all()
```

Comment:
How are we going to use this function? Also, we expect such functions to be paginated!

Reply:
Removed. Will add if it will be required in future (We were discussing about allowing exporting all the blog posts on the site so added it, but removed now)